### DISCUSSION TITLE
Requirements for theme-overrides at component level

### DISCUSSION THREAD

**Discussion by pac0rro** [tusen-ai/naive-ui#2362] [Q&A] [answered]
Hi all.

I wonder what are the requirements I must comply in order to get working **":theme-overrides"** at component level.

This piece of code doesn't work:
```
<n-input-number size="tiny" class="axisWeight" :bordered="false"  :theme-overrides="naiveStyles.axisWeightInput"/>
```

However, wrapping it in a **n-config-provider** tag works as expected.

``` 
<n-config-provider :theme-overrides="naiveStyles.axisWeightInput">
     <n-input-number size="tiny" class="axisWeight" :bordered="false" />
</n-config-provider>
```

I'm not developing under typescript, but the documentation points to an example coded in TS.

Using latest version 2.25.1.

Thanks in advance.

**Comment by 07akioni** [comment, marked as answer]
It should work. Please paste full `naiveStyles.axisWeightInput` here.

**Reply by 07akioni** [reply]
If you want to override a component, don't add key to it.

❌
`{ InputNumber: overridesObj }`

✅
`overridesObj`

**Reply by pac0rro** [reply]
Thanks, @07akioni.

This is the javascript that i've loaded as globalProperties:

``` 
export default {
    axisWeightInput: {
        color: "rgba(249, 0, 0, 0)",
        colorFocus: "#F81FA5EB",
        fontSizeTiny: "8px"
    }
};

```

**Reply by 07akioni** [reply]
> Thanks, @07akioni.
> 
> This is the javascript that i've loaded as globalProperties:
> 
> ```
> export default {
>     axisWeightInput: {
>         color: "rgba(249, 0, 0, 0)",
>         colorFocus: "#F81FA5EB",
>         fontSizeTiny: "8px"
>     }
> };
> ```

I don't think this is a valid global theme overrides.

**Reply by 07akioni** [reply]
It shouldn't work at both global level or component level

**Reply by pac0rro** [reply]
Yes, you are right. I have to change it when using **"n-config-provider"** tag to this:

```
export default {
    axisWeightInput: {
        Input: {
            color: "rgba(249, 0, 0, 0)",
            colorFocus: "#F81FA5EB",
            fontSizeTiny: "8px"
        }
    }
};
```
... in order to get it work.

Is there something I must activate to get it work at component level?

**Reply by 07akioni** [reply]
![image](https://user-images.githubusercontent.com/18677354/153045300-13128d88-76a4-4989-be93-465f151144e5.png)

**Reply by pac0rro** [reply]
Ohhhh, sorry. I didn't catch it. For nested components I must specify the styles with **"peers"**

This is the final config for component-level styles, following your advice:
```
export default {
    axisWeightInput: {
        peers: {
            Input: {
                color: "rgba(249, 0, 0, 0)",
                colorFocus: "#F81FA5EB",
                fontSizeTiny: "8px"
            }
        }
    }
};
```

Thank you very much!, @07akioni.

This UI framework is amazing!
